R/time_series.R

```diff
@@ -27,40 +28,30 @@ additive_classical_forecast <- function(series, lengths){
           y = deseasonalized[i:(i + len - 1)]
         )
       )
-
-    forecast <- predict.lm(model, newdata = list(t = i + len))
+      
+      seasonal_adjustment <- seasonality[i + len - frequency]
```

Comment:
```suggestion
      seasonal_adjustment <- seasonality[i + len]
```
Won't this fail for i = 1, len = 3 if frequency > 3?

Reply:
It hasn't been failing, but your actual point of "consider how this behaves for low i" is right. I'll investigate.

Reply:
Behavior is worse than failing, but turns out to be irrelevant. A negative index apparently removes that index from the vector, so that `seasonality[-3]` would return `seasonality[c(1:2,4:12}]`, applying everything other than the correct seasonal factor for those early points.